python/python/lancedb/remote/table.py

```diff
@@ -97,6 +97,8 @@ def index_stats(self, index_uuid: str):
     def create_scalar_index(
         self,
         column: str,
+        *,
+        index_type: Literal["BTREE", "BITMAP", "LABEL_LIST"] = "BTREE",
     ):
         """Creates a scalar index
         Parameters
```

Comment:
We need to make a decision around the python API at some point (doesn't have to be right now but this PR will make the problem worse).

If we keep adding items to the synchronous API and NOT the asynchronous API (as done here) then it makes the gap between the APIs larger and makes it harder for us to unify them.

The proper approach should be to either:
* Only add to the async API and add a PR for follow-up
* Add it to the async API and make the synchronous API call the async API (`table_names` on `LanceDBConnection` demonstrates this)

Reply:
Yeah, please add to the async API first. We need parity between these two because we need to transition the sync API to be a wrapper around the async API soon. Similar to Node, we don't want to have to keep maintaining two APIs.